client/me/account-password/index.jsx

```diff
@@ -22,113 +19,113 @@ import FormButton from 'calypso/components/forms/form-button';
 import FormButtonsBar from 'calypso/components/forms/form-buttons-bar';
 import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
 import FormInputValidation from 'calypso/components/forms/form-input-validation';
-import observe from 'calypso/lib/mixins/data-observe';
 import { errorNotice } from 'calypso/state/notices/actions';
 import { recordGoogleEvent } from 'calypso/state/analytics/actions';
 import { saveUserSettings } from 'calypso/state/user-settings/actions';
 import { isPendingPasswordChange } from 'calypso/state/user-settings/selectors';
+import { generatePassword } from 'calypso/lib/generate-password';
+import wp from 'calypso/lib/wp';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-/* eslint-disable react/prefer-es6-class */
-const AccountPassword = createReactClass( {
-	displayName: 'AccountPassword',
+const wpcom = wp.undocumented();
 
-	mixins: [ observe( 'accountPasswordData' ) ],
+class AccountPassword extends React.Component {
+	state = {
+		password: '',
+		validation: null,
+		pendingValidation: true,
+		isUnsaved: false,
+	};
 
 	componentDidMount() {
 		this.debouncedPasswordValidate = debounce( this.validatePassword, 300 );
-	},
+	}
 
 	componentDidUpdate( prevProps ) {
 		if ( prevProps.isPendingPasswordChange && ! this.props.isPendingPasswordChange ) {
 			// eslint-disable-next-line react/no-did-update-set-state
 			this.setState( { isUnsaved: false } );
 		}
-	},
+	}
 
-	componentWillUnmount() {
-		this.props.accountPasswordData.clearValidatedPassword();
-	},
-
-	getInitialState() {
-		return {
-			password: '',
-			pendingValidation: true,
-			isUnsaved: false,
-		};
-	},
-
-	generateStrongPassword() {
+	generateStrongPassword = () => {
 		this.setState( {
-			password: this.props.accountPasswordData.generate(),
+			password: generatePassword(),
 			pendingValidation: true,
 			isUnsaved: true,
 		} );
 		this.debouncedPasswordValidate();
-	},
-
-	validatePassword() {
-		debug( 'Validating password' );
-		this.props.accountPasswordData.validate(
-			this.state.password,
-			function () {
-				this.setState( { pendingValidation: false } );
-			}.bind( this )
-		);
-	},
+	};
+
+	validatePassword = async () => {
+		const password = this.state.password;
+
+		if ( '' === password ) {
+			this.setState( { validation: null, pendingValidation: false } );
+			return;
+		}
+
+		try {
+			const validationResult = await wpcom.me().validatePassword( password );
+
+			this.setState( { pendingValidation: false, validation: validationResult } );
+		} catch ( err ) {
+			this.setState( { pendingValidation: false } );
```

Comment:
If an error occurs, should we also alter `validation` state prop to ensure any prior state is being replaced? That would be important in the case where a password validation is successful, but then the user changes it and request fails for some reason (for example, because they went offline).

Reply:
In that case a user won't be able to submit the form which leads to a rather broken UI state. As we still validate the password on submit I'd keep it that way so the validation before submit is _nice to have_ but not mandatory.